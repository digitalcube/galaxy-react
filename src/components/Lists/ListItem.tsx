import React, { FC, PropsWithChildren, ReactNode } from 'react';
import { useMemo } from 'react';
import { StatusLabel, StatusLabelProps } from '../StatusLabel/StatusLabel';

export type ListItemProps = PropsWithChildren<{
  title: ReactNode;
  status?: StatusLabelProps;
  className?: string;
  rounded?: boolean;
  column?: boolean;
  alignItem?: 'start' | 'center' | 'end' | 'baseline' | 'stretch';
  justifyContent?: 'start' | 'end' | 'center' | 'between' | 'around';
  description?: ReactNode;
}>;

export const ListItemDescription: FC = ({ children }) => {
  if (!children) return null;
  return (
    <div className="mt-2 w-100">
      {typeof children === 'string' ? (
        <p className="mb-0 small">{children}</p>
      ) : (
        children
      )}
    </div>
  );
};

export const ListItem: FC<ListItemProps> = ({
  children,
  title,
  status,
  className,
  column,
  rounded,
  alignItem = 'center',
  justifyContent = 'between',
  description,
}) => {
  const classNames = useMemo(() => {
    const items = ['notification', 'd-flex', 'flex-wrap', className];
    if (column) items.push('flex-column');
    if (rounded) items.push('rounded');
    if (alignItem) items.push(`align-items-${alignItem}`);
    if (justifyContent) items.push(`justify-content-${justifyContent}`);
    return items.filter(Boolean);
  }, [className, rounded, alignItem, justifyContent, column]);
  return (
    <div className={classNames.join(' ')}>
      <div className="notification-status">
        {title} <StatusLabel {...status} />
      </div>
      {children}
      <ListItemDescription>{description}</ListItemDescription>
    </div>
  );
};
