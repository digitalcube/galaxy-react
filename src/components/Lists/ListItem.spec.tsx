import { render } from '@testing-library/react';
import React from 'react';
import { Default as ListItem } from './ListItem.stories';

describe('ListItem', () => {
  it('to match snapshot(default)', () => {
    const { asFragment } = render(<ListItem {...(ListItem.args as any)} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
