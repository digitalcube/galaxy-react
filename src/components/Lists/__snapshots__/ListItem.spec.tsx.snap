// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem to match snapshot(default) 1`] = `
<DocumentFragment>
  <div
    class="shifter-dashboard-columns pt-3"
  >
    <div
      class="base-notification-box rounded"
    >
      <div
        class="notification d-flex px-4 py-3 align-items-center justify-content-between"
      >
        <div
          class="notification-status"
        >
          Title 
        </div>
        children
      </div>
    </div>
  </div>
</DocumentFragment>
`;
